website/docs/basics/writing-styles.md

```diff
@@ -6,3 +7,122 @@ sidebar_label: Writing Styles
 
+Everyone says that writing styles are easy and usually don't think enough when doing so. Like everything in programming, if you don't, think the problems will add up exponentially, and you will have a headache fast.
+
+Here are some of our recommendations on writing our styles and how we manage everything to be as modular as possible.
+
+### Never use HEX color values in your files
+
+As is Bootstrap or any other frontend framework, you should never use hex color values in multiple places. You should assign that hex value to a variable and use that variable all over.
+
+You can read more about defining your colors in the [global settings](global-settings) chapter.
+
+### Never use the actual color name as a variable name
+
+You can name your variables how you like, but we recommend that you never use the actual color name as a variable name. It would be best if you used some generic variable name.
+
+For example, you can call blue color r2d2-color. With this approach, if for any reason your designer chooses to change the project color scheme at any point, you only need to change one variable value, and you are done. 
+
+We like to use [this site](https://chir.ag/projects/name-that-color) for our color names. Also, we recrement using a simple name as possible like:
+
+* Cornflower Blue will be `$base-corn-color` or `$base-flower-color` name.
+
+### All external variables must go to a components map
+
+We use maps for a lot of stuff in our boilerplate. The reason is that:
+
+* maps can contain multiple variables at one place
+* you don't need to write multiple variables at the top.
+* you can nest functionalities
+* you can have multiple values with the same name and just change the top-level map.
+* etc.
+
+We also have many helpers inside our [SASS Docs](/eightshift-docs/sass).
+
+In the component map, you should only add stuff that you will reuse in your component.
+
+**Examples:**
+
+A simple variable in a map.
+
+```scss
+$social-links: (
+  icon-color: global-settings(colors, black),
+);
+
+.social-links {
+  color: map-get-strict($social-links, icon-color);
+}
+```
+
+Nested variables in a map.
+
+```scss
+$social-links: (
+  colors: (
+    dark: global-settings(colors, black),
+    light: global-settings(colors, white),
+  ),
+  sizes: (
+    small: 10px,
+    big: 20px,
+  )
+);
+
+.social-links {
+  color: map-get-deep($social-links, colors, dark);
+
+  &--light {
+    color: map-get-deep($social-links, colors, light);
+  }
+
+  &__sizes {
+    font-size: map-get-deep($social-links, sizes, small);
+
+
+    &--big {
+      font-size: map-get-deep($social-links, sizes, big);
+    }
+  }
+}
+```
+
+### Never mix multiple maps in a single component
+
+You want all your styles to be continued to a single component and not depend on the external stuff. Sometimes this can't be avoided.
```

Comment:
```suggestion
You want all your styles to be contained to a single component and not dependent on the external components. Sometimes this can't be avoided.
```